### PR TITLE
[UI]: working on the feeds screen

### LINKED ISSUE #8: [UI] Create Feeds Screen
Create Feed Screen as per the guideline - [Feed Screen Design Adobe XD Link](https://xd.adobe.com/view/dd5d0245-b92b-4678-9d4a-48b3a6f48191-880e/screen/aabbe326-f4d9-4fae-9cbc-433491567613/specs/)

Follow contribution guidelines laid out in our [contribution guide](https://github.com/droidconKE/droidconKE2022ReactNative/blob/develop/CONTRIBUTING.md)

Screenshot 2022-09-12 at 21 14 34

### LINKED ISSUE #11: [ui] : Create Feed Screen
Create Feed Screen as per the guideline - [Feed Screen Design Adobe XD Link](https://xd.adobe.com/view/dd5d0245-b92b-4678-9d4a-48b3a6f48191-880e/screen/aabbe326-f4d9-4fae-9cbc-433491567613/specs/)
    Follow contribution guidelines laid out in our [contribution guide](https://github.com/droidconKE/droidconKE2022ReactNative/blob/develop/CONTRIBUTING.md)

**Additional context**
- Feed Screen
Screenshot 2022-09-12 at 23 13 03

### PR DESCRIPTION
**Pull request by mog-rn**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #11 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by mog-rn** [comment]
Still finalizing on the styling

**Comment by makunomark** [comment]
Hey @Mogakamo 
please expect the response from the api to look like this:
```json
{
    "data": [
        {
            "title": "Test",
            "body": "Good one",
            "topic": "droidconweb",
            "url": "https://droidcon.co.ke",
            "image": "http://localhost:8000/upload/event/feeds/dangyntvmaet8jgjpg.jpg",
            "created_at": "2020-03-19 18:45:49"
        },
        {
            "title": "niko fine",
            "body": "this is a test",
            "topic": "droidconweb",
            "url": "https://droidcon.co.ke",
            "image": null,
            "created_at": "2020-03-19 18:43:38"
        }
    ],
    "meta": {
        "paginator": {
            "count": 2,
            "per_page": "10",
            "current_page": 1,
            "next_page": null,
            "has_more_pages": false,
            "next_page_url": null,
            "previous_page_url": null
        }
    }
}
```

File: src/screens/FeedScreen.tsx [hunk 1]
```diff
@@ -1,35 +1,73 @@
...
+  },
+  {
+    id: 3,
+    description:
+      "Droidcon is a global conference series focused on the Android platform. It is organized by the Droidcon team, a group of Android enthusiasts from all over the world.",
+    image:
+      "https://droidconke.co.ke/wp-content/uploads/2020/09/IMG_20200919_120000.jpg",
+    postedAt: "2 days ago",
+  },
+];
```
Comment (brianwachira) [inline src/screens/FeedScreen.tsx]: Uhm, the expect the response from the api will look something like this:
```json
{
    "data": [
        {
            "title": "Test",
            "body": "Good one",
            "topic": "droidconweb",
            "url": "https://droidcon.co.ke",
            "image": "http://localhost:8000/upload/event/feeds/dangyntvmaet8jgjpg.jpg",
            "created_at": "2020-03-19 18:45:49"
        },
        {
            "title": "niko fine",
            "body": "this is a test",
            "topic": "droidconweb",
            "url": "https://droidcon.co.ke",
            "image": null,
            "created_at": "2020-03-19 18:43:38"
        }
    ],
    "meta": {
        "paginator": {
            "count": 2,
            "per_page": "10",
            "current_page": 1,
            "next_page": null,
            "has_more_pages": false,
            "next_page_url": null,
            "previous_page_url": null
        }
    }
}
```

So you can use this JSON instead

**Review by brianwachira — CHANGES_REQUESTED** [review]

File: src/screens/FeedScreen.tsx (hunk 1 above)
Comment (mog-rn) [reply src/screens/FeedScreen.tsx]: Ok let me work on it.

File: src/components/cards/FeedsCard.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      <Image
+        source={{
+          uri: image,
+        }}
+        style={styles.image}
+      />
+      <View style={styles.post_details}>
+        <TouchableHighlight style={styles.flex_row}>
+          <Text style={styles.share_btn}>Share</Text>
+          {/* <FeedIcon style={styles.feed_icon} /> */}
```
Comment (brianwachira) [inline src/components/cards/FeedsCard.tsx]: Add the correct share icon based on the [design](https://xd.adobe.com/view/dd5d0245-b92b-4678-9d4a-48b3a6f48191-880e/screen/aabbe326-f4d9-4fae-9cbc-433491567613/specs/).
- Download the icon in SVG format from the design
- Convert the SVG icon into a component using the link https://react-svgr.com/playground/?native=true&typescript=true
- Save it in the assets/icons folder. You can also check out how the other icon components have been done.

File: src/components/cards/FeedsCard.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    fontSize: 16,
+    alignItems: "center",
+  },
+  text: {
+    fontFamily: fonts.MONTSERRAT_REGULAR,
+    fontSize: 16,
+    color: colors.DROIDCONKE_BLACK,
+  },
+  image: {
+    width: 318,
```
Comment (brianwachira) [inline src/components/cards/FeedsCard.tsx]: The image is not spanning the full width of the screen
photo_2022-11-03_14-13-28

- You can set image width as a flexible property.
Change the width from `318` to  `100%` instead.

File: src/components/cards/FeedsCard.tsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  },
+  share_btn: {
+    color: colors.DROIDCONKE_BLUE,
+    fontSize: 16,
+    fontFamily: fonts.MONTSERRAT_BOLD,
+  },
+  time: {
+    color: colors.DROIDCONKE_DARK_GREY,
+    fontSize: 16,
+    fontFamily: fonts.MONTSERRAT_BOLD,
```
Comment (brianwachira) [inline src/components/cards/FeedsCard.tsx]: As per the [design](https://xd.adobe.com/view/dd5d0245-b92b-4678-9d4a-48b3a6f48191-880e/screen/aabbe326-f4d9-4fae-9cbc-433491567613/specs/)
fontSize and fontFamily should be 
`
    fontSize: 12,
    fontFamily: fonts.MONTSERRAT_REGULAR,
`

Please change accordingly

File: src/screens/FeedScreen.tsx [hunk 5]
```diff
@@ -1,35 +1,82 @@
...
+  return (
+    <SafeAreaView style={[styles.container, styles.paddingVertical]}>
+      <MainHeader />
+      <ScrollView style={styles.paddingHorizontal} nestedScrollEnabled={true}>
+        <FlatList
+          data={FEED[0].data}
+          renderItem={({ item }) => <FeedsCard {...item} />}
+          // keyExtractor={({ id }, index) => id}
+        />
+      </ScrollView>
```
Comment (brianwachira) [inline src/screens/FeedScreen.tsx]: So the flatlist is vertically scrollable by default. You can check out [React Native Docs on flatlist](https://reactnative.dev/docs/flatlist) for more information.
In that case, do away with the scrollview the define `styles.paddingHorizontal` on the flatlist.

File: src/screens/FeedScreen.tsx [hunk 6]
```diff
@@ -1,35 +1,82 @@
...
+        next_page: null,
+        has_more_pages: false,
+        next_page_url: null,
+        previous_page_url: null,
+      },
+    },
+  },
+];
+
+console.log(FEED[0].data);
```
Comment (brianwachira) [inline src/screens/FeedScreen.tsx]: You can clean out this console log.

**Review by brianwachira — CHANGES_REQUESTED** [review]

File: src/components/cards/FeedsCard.tsx [hunk 7]
```diff
@@ -54,7 +54,7 @@ const styles = StyleSheet.create({
     color: colors.DROIDCONKE_BLACK,
   },
   image: {
-    width: 318,
+    width: 100%,
```
Comment (brianwachira) [inline src/components/cards/FeedsCard.tsx]: image
- wrap `100%` on single quotes

**Review by brianwachira — CHANGES_REQUESTED** [review]

**Review by brianwachira — APPROVED** [review]

**Pull request merged**